hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/CertificateClient.java

```diff
@@ -126,6 +130,30 @@ X509Certificate getCertificate(String certSerialId)
    */
   Set<X509Certificate> getAllCaCerts();
 
+  /**
+   * This method serves as a convenience to solve the common problem in the
+   * code related to clients, and handle single SCM case and the data with
+   * which we provide one single cert to old clients.
+   * @return either the rootCA certificate, or the sub-CA certificate
+   *          of the node that served as the rootCA during bootstrap.
+   */
+  default String getSingleCACertificatePEM() throws SCMSecurityException {
+    return CertificateCodec.getPEMEncodedString(getSingleCACertificate());
+  }
+
+  /**
+   * This method serves as a convenience to solve the common problem in the
+   * code related to clients, and handle single SCM case and the data with
+   * which we provide one single cert to old clients.
+   * @return either the rootCA certificate, or the sub-CA certificate
+   *          of the node that served as the rootCA during bootstrap.
+   */
+  default X509Certificate getSingleCACertificate() {
```

Comment:
This method could possibly used in other places, (ReloadingTrustManager is what I know by heart, but it's going to change with Sammi's change), could you check once we get there, if there are other places we could use this?

Reply:
Actually after the new changes, I realized that this method is not needed anymore, and I should use the new methods that we have in the certificate client to get the list of rootCA certificates. I have reworked the relevant code, and related tests.